# Instrucciones

1. Determine el modelo de regresión lineal que le permita entender la relación entre la variable IPC y las variables regresoras PIB e Importaciones
2. ¿Qué puede concluir respecto de los estadísticos globales del modelo?
3. ¿Qué puede concluir respecto de los estadísticos individuales del modelo?

In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as ss
import statsmodels.formula.api as smf # Nos permite ajustar el modelo de regresión

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [ ]:
# Importamos la base de datos y la denominamos df por "data frame"
df = pd.read_csv('solemne_forma_4.csv', sep=';', decimal=",")

In [ ]:
df

IPC      PIB  Importaciones
0    53.8   1638.3          98185
1    56.9   1825.3         124228
2    60.6   2030.9         151907
3    65.2   2294.7           1762
4    72.6   2563.3           2127
5    82.4   2789.5         249750
6    90.9   3128.4          26567
7    96.5   3225.0         247642
8    99.6   3536.7         268901
9   103.9   3933.2         332418
10  107.6   4220.3          33888
11  109.6   4462.8         368425
12  113.6   4739.5         409765
13  118.3   5103.8         447189
14  124.0   5484.4         477665
15  130.7   5803.1         498438
16  136.2   5995.9          49120
17  140.3   6337.7         536528
18  144.5   6657.4         589394
19  148.2   7072.2         668690
20  152.4   7397.7         749374
21  156.9   7816.9         803113
22  160.5   8304.3         876470
23  163.0   8747.0         917103
24  166.6   9268.4         129980
25  172.2   9817.0        1224408
26  177.1  10128.0        1145900
27  179.9  10469.6        1164720
28  184.0  10960.8        1260717
29  188.9  11712.5        1472926
30  195.3  12455.8        1677371

In [ ]:
df.describe()

IPC           PIB  Importaciones
count   31.000000     31.000000   3.100000e+01
mean   127.490323   6126.464516   5.485378e+05
std     42.251693   3178.399420   4.714482e+05
min     53.800000   1638.300000   1.762000e+03
25%     98.050000   3380.850000   1.409435e+05
50%    130.700000   5803.100000   4.471890e+05
75%    161.750000   8525.650000   8.397915e+05
max    195.300000  12455.800000   1.677371e+06

# Solución

1.

In [ ]:
# Usamos el método de los mínimos cuadrados para ajustar la regresión lineal multiple
Modelo = smf.ols('IPC ~ PIB+Importaciones', df).fit() #Ajustamos el modelo
# Usamos summary para pedir el resultado del modelo
Modelo.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    IPC   R-squared:                       0.965
Model:                            OLS   Adj. R-squared:                  0.963
Method:                 Least Squares   F-statistic:                     388.9
Date:                Thu, 18 Mar 2021   Prob (F-statistic):           3.74e-21
Time:                        02:50:49   Log-Likelihood:                -107.46
No. Observations:                  31   AIC:                             220.9
Df Residuals:                      28   BIC:                             225.2
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=================================================================================
                    coef    std err          t      P>|t|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept        44.2386      3.745     11.811      0.000      36.567      51.911
PIB               0.0147      0.001     13.872      0.000       0.013       0.017
Importaciones -1.273e-05   7.16e-06     -1.779      0.086   -2.74e-05    1.93e-06
==============================================================================
Omnibus:                        7.364   Durbin-Watson:                   0.325
Prob(Omnibus):                  0.025   Jarque-Bera (JB):                3.290
Skew:                          -0.529   Prob(JB):                        0.193
Kurtosis:                       1.806   Cond. No.                     1.84e+06
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
[2] The condition number is large, 1.84e+06. This might indicate that there are
strong multicollinearity or other numerical problems.
"""

1. El modelo obtenido es $IPC = 44.2386+0.0147*PIB-1.273exp^{-5}*Importaciones$

2. Análisis estadísticos globales. A partir de los valores de F, R2 y R2 Adj podemos concluir que el modelo es capaz de explicar cerca del 96% de la varianza. F es significativo ($F_{est} < pvalor$). Luego, se rechaza la hipótesis nula de que los coeficientes de regresiones son iguales a cero, por lo que al menos un coeficiente debe ser distinto de cero.

3. Los coeficientes asociados a la variable PIB mantiene un test-t estadísticamente significativo ($ t_{est} < pvalor$), por lo que se rechaza la hipótesis nula para esta variable. Mientras que para la variable Importaciones aceptamos la hipótesis nula, Es decir, su coeficiente de regresión es igual a cero.
